core/src/main/java/cucumber/runner/PickleTestStep.java

```diff
@@ -50,7 +50,7 @@ Result run(EventBus bus, String language, Scenario scenario, boolean skipSteps)
         results.add(super.run(bus, language, scenario, skipNextStep));
 
         for (HookStep after : afterStepHookSteps) {
-            results.add(after.run(bus, language, scenario, skipSteps));
+            results.add(after.run(bus, language, scenario, skipNextStep));
```

Comment:
Why did you make this change? I would always expect an after step hook to run so it can clean up anything setup by a before step.

Reply:
And that should have been captured in a test I guess.

Reply:
@mpkorstanje, if beforeStep fails, step is skipped but afterStep runs, is that a correct behaviour? for eg, I was taking a snapshot in after step and it appeared in the report even if the actual step was skipped from executing

Reply:
Yes. There may be multiple before steps. If a step other then the failing one allocates some resource then it should be deallocated in an after step. Skipping the after step execution will leave that resource allocated and may cause a leak.